base/ca/src/com/netscape/ca/CAService.java

```diff
@@ -1374,6 +1383,43 @@ boolean verifySCT(CTResponse response, byte[] tbsCert, String logPublicKey) {
         }
     }
 
+    /**
+     * parses and gleans the hashing algorithm from the returned
+     * CT signature
+     * cfu
+     */
+    /*
+       enum HashAlgorithm {
+            none, md5, sha1, sha224, sha256, sha384, sha512};
+    */
+    enum HashAlgorithm {none, MD5, SHA1, SHA224, SHA256, SHA384, SHA512};
+    public String getHashAlgFromSig(byte[] ct_signature) {
+
+        int hashingAlg = Byte.toUnsignedInt(ct_signature[0]);
+        if (hashingAlg < 4 || hashingAlg > 5)
+            return null;
+
+        return HashAlgorithm.values()[hashingAlg].name();
```

Comment:
`.name()` seems redundant? Can it be simple like this: `HashAlgorithm.values()[hashingAlg]` ?

Reply:
I think that was the first thing I tried.  It gave me the following error:
com/netscape/ca/CAService.java:1402: error: incompatible types: HashAlgorithm cannot be converted to String
        return HashAlgorithm.values()[hashingAlg];
                                     ^
com/netscape/ca/CAService.java:1420: error: incompatible types: SignatureAlgorithm cannot be converted to String
        return SignatureAlgorithm.values()[signingAlg];

Reply:
weird. But, ok